### PR TITLE
issue #2122 Add ability to parse raw text into docvectors on-the-fly for impact indexes

### PR DESCRIPTION
**Pull request by AileenLin**

### CODE REVIEW

File: src/main/java/io/anserini/search/SimpleImpactSearcher.java [hunk 1]
```diff
@@ -97,12 +104,23 @@ protected SimpleImpactSearcher() {
   }
 
   /**
-   * Creates a {@code SimpleImpactSearcher}.
+   * Creates a {@code SimpleSearcher}.
```
Comment (lintool) [inline src/main/java/io/anserini/search/SimpleImpactSearcher.java:107]:
```suggestion
   * Creates a {@code SimpleImpactSearcher}.
```

**Comment by lintool** [comment]
We have an error getting test coverage: https://app.codecov.io/gh/castorini/anserini/pull/2133

File: src/main/java/io/anserini/rerank/RerankerContext.java [hunk 2]
```diff
@@ -46,6 +47,20 @@ public RerankerContext(IndexSearcher searcher, K queryId, Query query, String qu
     this.queryTokens = queryTokens;
     this.filter = filter;
     this.searchArgs = searchArgs;
+    this.weightedQueryTokens = null;
+  }
+
+  public RerankerContext(IndexSearcher searcher, K queryId, Query query, String queryDocId, String queryText,
```
Comment (lintool) [inline src/main/java/io/anserini/rerank/RerankerContext.java:53]: hi @AileenLin can you explain why we need to add this code path?

`SearchCollection` is able to perform Rocchio/RM3 with impact exists, right?
E.g., https://github.com/castorini/anserini/blob/master/src/main/java/io/anserini/search/SearchCollection.java#L1152

Comment (AileenLin) [reply src/main/java/io/anserini/rerank/RerankerContext.java:53]: the SimpleImpact search method take in a Map of token frequency, we can either add the weitedQueryToken here, or add another search method to process text in SimpleImpact. But if for pretokenized Map of token frequency, I prefered the second way to do it

Comment (lintool) [reply src/main/java/io/anserini/rerank/RerankerContext.java:53]: In that case, would it make sense for `SearchCollection` and `SimpleImpactSearcher` to use the same code path?

**Pull request closed without merging**